### ISSUE TITLE
Linux - Flatpak build

### ISSUE THREAD

**Issue by LeandroStanger** [FlashpointProject/launcher#94]
Please, consider supporting [Flatpak](http://docs.flatpak.org/en/latest/introduction.html#reasons-to-use-flatpak) and publishing BlueMaxima’s Flashpoint Infinity on [Flathub](https://github.com/flathub/flathub/wiki/App-Submission).

**Comment by patriciaa260** [comment]
i agree, please make flatpak version

**Comment by Kallb123** [comment]
I can't see this on Flathub, can we reopen the issue?

**Comment by LarryIsBetter** [comment]
Hello Id also love to see this, it'd make installing easier!

**Comment by WMan22** [comment]
Just got a steam deck and would love to use this on it, a flatpak would solve a lot of issues.

**Comment by double-daruma** [comment]
Does anyone know how the docker images linked on https://bluemaxima.org/flashpoint/datahub/Linux_Support are created? I suppose those could be used as a starting point for how to flatpak this?

**Comment by Swizzler121** [comment]
So, if the launcher gets ported, is everything else good behind that? It seems fairly trivial to do so, it's built in GTK and is using electron and NPM... I'm just confused as to why this hasn't been attempted yet so is there some other component that needs packaged alongside the launcher i'm missing? If not I might give it a shot (no promises, i've only built snaps before but snap sucks and I wanted a project to try learning to build a flatpak on and this looks ripe for it)

**Comment by Twig6943** [comment]
bump

**Comment by bojogon7** [comment]
Flatpak doesn't need to be Linux port it can be wine version like https://snapcraft.io/notepad-plus-plus (I don't know any flathub app which does it) 

It would run in the same runtime and version of wine/proton on every distro, which would make installation on Linux very easy. Current methods for running it on linux on wiki. 

- Download the Windows version of Flashpoint and run it in a Windows virtual machine. This is the recommended non-native option.
- Run the Windows version of Flashpoint with Wine. This may or may not work well depending on the exact build of Wine used

Flathub policy: 
"Software using emulation or translation layers

Windows software submissions that are using Wine or any submissions that aren't native to the Linux desktop and is using some emulation or translation layer will only be accepted if they are submitted officially by upstream with the intention of maintaining it in an official capacity."
Source: https://docs.flathub.org/docs/for-app-authors/requirements#software-using-emulation-or-translation-layers

And if there is bigger interest in making native Linux client wine port can be later replaced.

**Comment by Mertsudo** [comment]
bump

**Comment by RubberDuckyDJ** [comment]
This would be so nice.

**Comment by rcfox2** [comment]
bump

**Comment by Twig6943** [comment]
Work in progress appimage https://github.com/Twig6943/launcher/actions/runs/31216826555/job/92992204493

Please test and let me know what works and what doesn't

I know this is the flatpak issue but I see it relevant enough to mention

**Comment by rcfox2** [comment]
> Work in progress appimage https://github.com/Twig6943/launcher/actions/runs/31216826555/job/92992204493
> 
> Please test and let me know what works and what doesn't
> 
> I know this is the flatpak issue but I see it relevant enough to mention

I wish you good luck finding someone to test it, but I have doubts that the same people asking for a flatpak will want to run this on their system uncompartmentalized

**Comment by MateusRodCosta** [comment]
Just a tip for whoever wants a Flatpak.

From my point of view and the info I have on flatpaks and Flashpoint I can say:

* The biggest issue here seems to be that Flashpoint is created with the goal of running self contained from a single folder, whereas flatpaks have a read only /app folder
* To actually work around this you likely want to figure out the folders Flashpoint relies on writing to and redirect them, all while the app is running
* Normally on a non Flatpak install we probably would want to have a `~/.local/share/Flashpoint` folder we redirect these flowers to, but for Flatpak we likely want `~/.var/app/<app id>/data/Flashpoint`
*There's also the extension points issue. You likely will want to split some features into extension points and you likely want to take advantage of the already shipped wine on Flathub.
* I do not know precisely how you would handle the potential lack of Flashpoint Manager, but to trigger install of extensions or updates of the flatpaks you usually have to use the flatpak tool or use custom code that communicate with it

To sum up it's likely very possible but might require some coordinating it with the Flashpoint team

**Comment by Twig6943** [comment]
> Just a tip for whoever wants a Flatpak.
> 
> From my point of view and the info I have on flatpaks and Flashpoint I can say:
> 
> * The biggest issue here seems to be that Flashpoint is created with the goal of running self contained from a single folder, whereas flatpaks have a read only /app folder
> * To actually work around this you likely want to figure out the folders Flashpoint relies on writing to and redirect them, all while the app is running
> * Normally on a non Flatpak install we probably would want to have a `~/.local/share/Flashpoint` folder we redirect these flowers to, but for Flatpak we likely want `~/.var/app/<app id>/data/Flashpoint`
>   *There's also the extension points issue. You likely will want to split some features into extension points and you likely want to take advantage of the already shipped wine on Flathub.
> * I do not know precisely how you would handle the potential lack of Flashpoint Manager, but to trigger install of extensions or updates of the flatpaks you usually have to use the flatpak tool or use custom code that communicate with it
> 
> To sum up it's likely very possible but might require some coordinating it with the Flashpoint team

Ive done plenty of flatpaks as well and this is kinda why i went with appimage first. The app needs to be switched up a bit like you said in order to be flatpak. There seems to be 0 linux maintainers for this at the moment (the previous one retired) soo if somebody decides to pick up the work hit me up I can do the flatpak part

Also justs a reminder that appimage package managers such as [soar](https://github.com/pkgforge/soar) & [am](https://github.com/ivan-hc/AM) exist (not a whole lot of people know about them sadly)
